sdcm/nemesis.py

```diff
@@ -5008,6 +5011,115 @@ def disrupt_disable_binary_gossip_execute_major_compaction(self):
+
+            with self.cluster.cql_connection_exclusive(node=self.target_node) as session:
+                try:
+                    stmt = SimpleStatement("INSERT INTO banned_keyspace.table1 (id, name) VALUES (1, 'name1');",
+                                           consistency_level=ConsistencyLevel.ALL)
+                    session.execute(stmt)
+                    raise BannedQueryExecutedSuccessful(
+                        "Query from banned node was executed succesful with Consistency.ALL")
+                except BannedQueryExecutedSuccessful:
+                    self.log.error("Banned query passed to cluster from banned node")
+                    raise
+                except (NoHostAvailable, OperationTimedOut, Unavailable) as exc:
+                    self.log.debug("Query failed with error: %s as expected", exc)
+
+        except Exception as exc:  # pylint: disable=broad-except
+            self.log.error("Method failed with error %s", exc)
+            raise
+        finally:
+            # because target node removed from cluster, terminate it
+            # and bootstrap new one
+            LOGGER.debug("Terminating node %s", self.target_node.name)
+            self.cluster.terminate_node(self.target_node)
+            # check node was removed previously, if not remove it from cluster
+            removed_node_status = self.cluster.get_node_status_dictionary(
+                ip_address=self.target_node.ip_address, verification_node=working_node)
+            if removed_node_status is not None:
+                working_node.run_nodetool(f"removenode {target_host_id}", retry=3)
+            LOGGER.debug("Restore number of cluster nodes")
+            new_node = self._add_and_init_new_cluster_node()
+            LOGGER.debug("New node %s was added", new_node.name)
+
+            with DbEventsFilter(db_event=DatabaseLogEvent.RUNTIME_ERROR,
+                                line="failed to repair"):
+                for node in self.cluster.nodes:
+                    try:
+                        self.repair_nodetool_repair(node=node, publish_event=False)
+                    except Exception as details:  # pylint: disable=broad-except
+                        self.log.error(f"failed to execute repair command "
+                                       f"on node {node} due to the following error: {str(details)}")
+
+            try:
+                with self.cluster.cql_connection_patient(new_node) as session:
+                    LOGGER.debug("Check create keyspace is empty")
+                    result = list(session.execute("SELECT * from banned_keyspace.table1"))
+                    LOGGER.debug("Query result %s", result)
+                    assert not result, f"New rows were added from banned node, {result}"
+            finally:
+                with self.cluster.cql_connection_patient(working_node) as session:
+                    LOGGER.debug("Drop create keyspace")
```

Comment:
In the `Drop create keyspace` statement I see 2 commands `drop` and then `create` a keyspace.
But according to the code below it is not correct assumption.
So, better to rephrase it to avoid misunderstandings.

Reply:
updated